### PR TITLE
Gist - card 6

### PR DESCRIPTION
**Pull request by danacastroman**

### CODE REVIEW

File: gist_content.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+require 'pathname'
+
+class Received_values
+  attr_reader :data
+
+  def initialize(pathname, description, state)
+    @pathname = pathname
+    @description = description
+    @state = state
+    @content = nil
```
Comment (Joelit0) [inline gist_content.rb]: Is it necessary to set nil content?

File: gist_content.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  end
+  
+  def read_path
+    path = Dir[@pathname + "/**/*"]
+    path.each do |x|
+      @filename = File.basename(x)
+      @content = File.read(x)
+      @data["files"][@filename] = content = {'content' => @content} 
+    end                  
+    # p @data
```
Comment (pazcuturi) [inline gist_content.rb]: Can we remove this comment?

**Review by pazcuturi — APPROVED** [review]

File: gist_content.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    }
+    self.read_path
+  end
+  
+  def read_path
+    path = Dir[@pathname + "/**/*"]
+    path.each do |x|
+      @filename = File.basename(x)
+      @content = File.read(x)
+      @data["files"][@filename] = content = {'content' => @content}
```
Comment (Joelit0) [inline gist_content.rb]: Use spaces before and after the {}. Example: { 'Hello' => 'World' }

File: main.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      p "Error de conexión"
+      p "Intentar de nuevo?"
+      intentar_de_nuevo = gets.chomp.capitalize
+
+      if intentar_de_nuevo == "No"
+        break
+      end
+    else
+      if request.response.code == "201"  
+        break
```
Comment (Joelit0) [inline main.rb]: Use "break if" for better organization in your code: http://www.java2s.com/Code/Ruby/Statement/Breakif.htm

File: request.rb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    http = Net::HTTP.new(uri.host, uri.port)
+    http.use_ssl = true
+    http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+
+    request = Net::HTTP::Post.new(uri.request_uri)
+    Dotenv.load
+    request['Authorization'] = "token #{ENV['GITHUB_ACCESS_TOKEN']}"
+    request.body = @data.to_json
+
+    @response = http.request(request)
```
Comment (Joelit0) [inline request.rb:30]: Why is '@response' an instance variable?

File: main.rb [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  if values.data
+    values.body
+    request = Assemble.new(description, @state, values.data)
+
+    begin
+      request.request
+    rescue SocketError => exception
+      p "Error de conexión"
+      p "Intentar de nuevo?"
+      intentar_de_nuevo = gets.chomp.capitalize
```
Comment (Joelit0) [inline main.rb]: Maybe you can name your variables in English as there are other variables in English.

File: main.rb [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+  if values.data
+    values.body
+    request = Assemble.new(description, @state, values.data)
+
+    begin
+      request.request
+    rescue SocketError => exception
+      p "Error de conexión"
+      p "Intentar de nuevo?"
```
Comment (Joelit0) [inline main.rb]: The same comment with the 'put', make your code in English or Spanish but not both

File: gist_content.rb [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  attr_reader :data
+
+  def initialize(pathname, description, state)
+    @pathname = pathname
+    @description = description
+    @state = state
+  end
+
+  def path_existence
+    if Dir.exist?(@pathname)
```
Comment (Joelit0) [inline gist_content.rb]: Maybe you should use the ternary operator for better organization. 
https://www.rubyguides.com/2019/10/ruby-ternary-operator/

File: .gitignore [hunk 9]
```diff
@@ -0,0 +1,2 @@
+.DS_Store.
+.env
```
Comment (jrgonh) [inline .gitignore]: Newline at end of file

**Review by jrgonh — CHANGES_REQUESTED** [review]

**Review by jrgonh — APPROVED** [review]

**Pull request merged**